graylog2-server/src/main/java/org/graylog2/plugin/inject/Graylog2Module.java

```diff
@@ -471,8 +475,22 @@ protected void registerJacksonSubtype(Class<?> klass, String name) {
 
     private static class ExceptionMapperType extends TypeLiteral<Class<? extends ExceptionMapper>> {}
 
-    protected void registerRestControllerPackage(String packageName) {
-        final Multibinder<RestControllerPackage> restControllerPackages = Multibinder.newSetBinder(binder(), RestControllerPackage.class);
-        restControllerPackages.addBinding().toInstance(RestControllerPackage.create(packageName));
+    /**
+     * Adds given API resource as a system resource. This should not be used from plugins!
+     * Plugins should use {@link org.graylog2.plugin.PluginModule#addRestResource(Class)} instead to ensure the
+     * addition of the path prefix.
+     *
+     * @param restResourceClass the resource to add
+     */
+    protected void addSystemRestResource(Class<?> restResourceClass) {
```

Comment:
Would it make sense to extend `Graylog2Module` with something like `CoreModule` that `PluginModule` does not extend from and put `addSystemResource` in there? I am trying to avoid that plugin authors make use of this methods because it's available and e.g. suggested in the IDE.

Reply:
Yeah, that's a good point.

`PluginModule` currently extends `Graylog2Module` and lots of modules in the server extend `Graylog2Module`. We also have several modules in the server which extend `PluginModule`.

So we would need to introduce a `CoreModule` that extends `Graylog2Module` and then switch all server modules that extend `PluginModule` over to `CoreModule` to get access to the API resource binding helper. That means we probably have to push down some helpers from `PluginModule` to `Graylog2Module`, depending on what the server modules that currently extend `PluginModule` need.

Correct?

Sounds like a separate PR. :smile:

Reply:
Alright, I am fine with doing that in a separate Pr.